packages/react-server/src/ReactFizzServer.js

```diff
@@ -498,24 +498,28 @@ function renderSuspenseBoundary(
   boundarySegment.status = COMPLETED;
   boundarySegment.children.push(innerSegment);
 
-  // We create suspended task for the fallback because we don't want to actually work
-  // on it yet in case we finish the main content, so we queue for later.
-  const suspendedFallbackTask = createTask(
-    request,
-    fallback,
-    parentBoundary,
-    innerSegment,
-    fallbackAbortSet,
-    task.legacyContext,
-    task.context,
-    null,
-  );
-  if (__DEV__) {
-    suspendedFallbackTask.componentStack = task.componentStack;
+  // if avoidThisFallback is set to true then we act as though there is no fallback in SSR
+  // Instead, we let the client render the fallback if the component suspends there.
+  if (props.unstable_avoidThisFallback !== true) {
```

Comment:
So I think adding it here works but in a kind of round about way.

It only works because not rendering the fallback is the same as suspending the fallback. So it’ll prefer the parent until the main content completes and if that happens we tend to prefer the main content.

There might be a more straightforward way of doing this with less overhead though.

Reply:
I'll rethink this.

Reply:
Eventually we’ll want this to not be a prop but a complement type. In that case we’d want to fork earlier.

One technique we often use it to copy the code path. I.e. avoid dry. Then remove as much as we can to specialize. That could be an interesting exercise here.